harness/determined/_trial_controller.py

```diff
@@ -106,6 +106,11 @@ def supports_mixed_precision(cls: Type["TrialController"]) -> bool:
     def supports_averaging_training_metrics(cls: Type["TrialController"]) -> bool:
         return False
 
+    @classmethod
+    def get_metric_writer(cls: Type["TrialController"]) -> tensorboard.BatchMetricWriter:
+        # default to the original kludge to avoid regression in custom trial controllers
+        return tensorboard.get_metric_writer()
+
```

Comment:
I don't think there is a single custom TrialController in existence (outside of our own test fixtures).

If there is, they are super-hackers since they never once asked us for help, and as super-hackers they can figure this out on their own.

That said, if you want to leave it in "just in case" that's ok, but if you get rid of `TrainContext.set_tensorboard_metric_writer()` (which you absolutely should) you don't need this anymore.